prereise/gather/winddata/hrrr/hrrr.py

```diff
@@ -0,0 +1,19 @@
+from prereise.gather.winddata.hrrr.constants import HRRR_S3_BASE_URL
+from prereise.gather.winddata.hrrr.downloader import Downloader
+from prereise.gather.winddata.hrrr.hrrr_api import HrrrApi
+
+
+def retrieve_data(start_dt, end_dt, directory):
+    """Retrieves all HRRR wind data for all hours between start_dt and
+    end_dt. (In a future PR) will convert all that wind data to
+    Pout in order to be compatible with REISE.
+
+    Sample usage:
+    retrieve_data(start_dt=datetime.fromisoformat("2016-01-01"), end_dt=datetime.fromisoformat("2016-01-01") + timedelta(hours=2), directory="./")
+
+    :param datetime.datetime start_dt: datetime to start at
+    :param datetime.datetime end_dt: datetime to end at
+    :param str directory: file directory to download data into
+    """
+    api = HrrrApi(Downloader, HRRR_S3_BASE_URL)
```

Comment:
I noticed that we have four URLs in `constants.py` but only this one is used in the code. Are the other three references of backups? Do we want the user has the capability of overriding the URL here?

Reply:
Also, it would be nice to fall back on the other ones if some reason the selected one (hard coded S3 one or user defined as @BainanXia proposed) is unavailable.

Reply:
I'm going to punt this to the next PR, primarily because this function isn't complete. The 4 hard coded urls are there more for documentation for code and so any user who wants to manually use the API (say in a jupyter notebook) can use other options